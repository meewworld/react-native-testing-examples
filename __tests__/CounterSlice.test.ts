import counterReducer, {
  initialState,
  increment,
  decrement,
} from '../src/redux/counterSlice';

describe('CounterSlice reducer', () => {
  it('should return the initial state on first run', () => {
    // @ts-ignore
    expect(counterReducer(undefined, {})).toMatchSnapshot();
  });
  it('correctly increments counter value on action "increment"', () => {
    expect(counterReducer(initialState, increment())).toMatchSnapshot();
  });

  it('correctly decrements counter value on action "decrement"', () => {
    expect(counterReducer(initialState, decrement())).toMatchSnapshot();
  });
});
