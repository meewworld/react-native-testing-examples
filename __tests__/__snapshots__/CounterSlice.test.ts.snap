// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterSlice reducer correctly decrements counter value on action "decrement" 1`] = `
Object {
  "value": -1,
}
`;

exports[`CounterSlice reducer correctly increments counter value on action "increment" 1`] = `
Object {
  "value": 1,
}
`;

exports[`CounterSlice reducer should return the initial state on first run 1`] = `
Object {
  "value": 0,
}
`;
